frontend/packages/kubevirt-plugin/src/components/create-vm-wizard/redux/state-update/vm-settings-tab-state-update.ts

```diff
@@ -162,6 +167,59 @@ const osUpdater = ({ id, prevState, dispatch, getState }: UpdateOptions) => {
   if (!isWindows && windowsTools) {
     dispatch(vmWizardInternalActions[InternalActionType.RemoveStorage](id, windowsTools.id));
   }
+
+  const iUserTemplates = iGetLoadedCommonData(state, id, VMWizardProps.userTemplates);
+  const iCommonTemplates = iGetLoadedCommonData(state, id, VMWizardProps.commonTemplates);
+  const template = toShallowJS(iGetRelevantTemplate(iUserTemplates, iCommonTemplates, selectors));
```

Comment:
toShallowJS is an expensive operation (we should avoid such operations in state update functions if we can since they can get called frequently)

and we need the template just for one annotation. I see that we are searching them by prefix. So maybe we can call toShallowJS on just the annotations only?

Reply:
also it doesn't make sense to go through all the relevant templates if os is empty (resulting template is random). We should just clear out the `VMSettingsField.CLONE_BASE_DISK_IMAGE` value in that case

Reply:
fixed :+1: